### PR TITLE
Add application frontend

### PR DESCRIPTION
**Pull request by michael-markl**
This PR adds a structure which allows to create forms relatively easily. The PR includes
* an application form for the blue EAK using the most common entitlement type ("Berechtigungsgrund"): Work at Organization(s) 
* auto-saving the application form (including files)
* Send GraphQL request to backend (including files)
* request gets processed successfully in the backend and is visible in the administration tool.

What's left:
* Make it a bit prettier
* #624: Query regionId from PLZ of the antragsteller
* #628: Add options for JuLeiCa, Feuerwehrler, and goldene EAK
* #629: Check whether the form itself is complete (content wise)
* #622: Validate inputs and files in the backend
* #329: Check files for malware (or sandboxing by browser?)
* #627: Add possibility for organizations to "verify" an application entitlement
* #626: Make the application frontend a "standalone app" by e.g. adding an entry point in the `administration` folder. 
* #625: Add screen for after the application is submitted.
* Properly handle dates as a separate JsonField (done in #622)

Sorry for the large PR, half of it is package-lock.json :)

### CODE REVIEW

File: administration/package.json [hunk 1]
```diff
@@ -8,13 +8,21 @@
     "@blueprintjs/datetime": "^4.3.12",
     "@blueprintjs/popover2": "^1.7.3",
     "@blueprintjs/select": "^4.6.6",
+    "@emotion/react": "^11.10.4",
+    "@emotion/styled": "^11.10.4",
+    "@fontsource/roboto": "^4.5.8",
+    "@mui/icons-material": "^5.10.9",
+    "@mui/lab": "^5.0.0-alpha.104",
+    "@mui/material": "^5.10.9",
```
Comment (maxammann) [inline administration/package.json:16]: Haven't reviewed the rest of the PR: Do we need Material? Isn't blueprint enough?

I made pleasant experience with that UI framework: https://chakra-ui.com/

Comment (michael-markl) [reply administration/package.json:16]: Blueprint is unfortunately focused on desktop environments. @mui was the next best thing that came to mind with good mobile support.

**Comment by maxammann** [comment]
This is ready for review, right?

**Comment by michael-markl** [comment]
@maxammann yup

File: administration/src/application/components/primitive-inputs/DateForm.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+type AdditionalProps = { label: string; minWidth?: number }
+const dateForm: Form<DateFormState, Options, ValidatedInput, AdditionalProps> = {
+  initialState: { type: 'DateForm', value: '' },
+  getValidatedInput: ({ value }) => {
+    if (value === '') return { type: 'error', message: 'Feld ist erforderlich.' }
+    const date = Date.parse(value)
+    if (isNaN(date)) {
+      return { type: 'error', message: 'Eingabe ist kein gültiges Datum.' }
+    }
+    return { type: 'valid', value: new Date(date).toString() }
```
Comment (maxammann) [inline administration/src/application/components/primitive-inputs/DateForm.tsx:17]:
```suggestion
    return { type: 'valid', value: new Date(date).toLocaleDateString() }
```

https://caniuse.com/date-tolocaledatestring

File: administration/src/application/globalArrayBuffersManager.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  addArrayBuffer(arrayBuffer: ArrayBuffer): number {
+    const newKey = Math.max(...this.arrayBuffers.map(({ key }) => key), 0) + 1
+    this.arrayBuffers.push({ key: newKey, value: arrayBuffer })
+    localforage.setItem(globalArrayBuffersKey, this.arrayBuffers)
+    return newKey
+  }
+
+  removeArrayBufferByKey(key: number): void {
```
Comment (maxammann) [inline administration/src/application/globalArrayBuffersManager.ts:31]: Remove this function and replace with `cleanupGarbageBuffers` (implement getFiles funciton for every function)

**Review by maxammann — APPROVED** [review]

File: administration/src/application/components/primitive-inputs/DateForm.tsx (hunk 2 above)
Comment (michael-markl) [reply administration/src/application/components/primitive-inputs/DateForm.tsx:17]: Leaving this for now as it is. Dates need to be supported by the backend first.

**Comment by michael-markl** [comment]
@maxammann You might want to have another look. If everything's okay, feel free to merge.

**Pull request merged**